pandas/io/xml.py

```diff
@@ -345,21 +345,7 @@ def _iterparse_nodes(self, iterparse: Callable) -> list[dict[str, str | None]]:
                     row = {}
 
             if row is not None:
-                if self.names:
-                    for col, nm in zip(self.iterparse[row_node], self.names):
-                        if curr_elem == col:
-                            elem_val = elem.text.strip() if elem.text else None
-                            if row.get(nm) != elem_val and nm not in row:
-                                row[nm] = elem_val
-                        if col in elem.attrib:
-                            if elem.attrib[col] not in row.values() and nm not in row:
-                                row[nm] = elem.attrib[col]
-                else:
-                    for col in self.iterparse[row_node]:
-                        if curr_elem == col:
-                            row[col] = elem.text.strip() if elem.text else None
-                        if col in elem.attrib:
-                            row[col] = elem.attrib[col]
+                self._iterparse_nodes_action(curr_elem, row, row_node, elem)
```

Comment:
Thanks but as is I'm -1 on this change. As it, code just moved (and added 1 layer of indirection with a function call) so I don't find this any simpler to understand

Reply:
Thanks for the review. What i had in mind when doing these changes was that this function, along with some others, looked quite daunting for it's size and the amount of conditionals one right after another. So I only did a simple extract method to just take some load off the function itself. I'll see what I can do more to improve the readability of this function, but it might be hard to actually make it simpler insteed of accidentally making it more complex.

Reply:
I agree with @mroeschke, this is actually harder to read now. Having the conditionals all together removes some cognitive load. If there is no way to simplify the actual execution logic, I would prefer keeping it as is